analyzers/tests/SonarAnalyzer.UnitTest/TestCases/InfiniteRecursion.RoslynCfg.cs

```diff
@@ -0,0 +1,398 @@
+﻿using System;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Tests.Diagnostics
+{
+    class InfiniteRecursion
+    {
+        int Pow(int num, int exponent)   // Noncompliant; no condition under which pow isn't re-called
+//          ^^^
+        {
+            num = num * Pow(num, exponent - 1);
+            return num;  // this is never reached
+        }
+
+        void Test1() // Noncompliant {{Add a way to break out of this method's recursion.}}
+        {
+            var i = 10;
+            if (i == 10)
+            {
+                Test1();
+            }
+            else
+            {
+                switch (i)
+                {
+                    case 1:
+                        Test1();
+                        break;
+                    default:
+                        Test1();
+                        break;
+                }
+            }
+        }
+
+        void Test2()  // Noncompliant
+        {
+            var i = 10;
+            switch (i)
+            {
+                case 1:
+                    goto default;  // Missing secondary location
+                default:
+                    goto case 1;   // Missing secondary location
+            }
+        }
+
+        void Test3()
+        {
+            var i = 10;
+            switch (i)
+            {
+                case 1:
+                    goto default;
+                case 2:
+                    break;
+                default:
+                    goto case 1; // FN
+            }
+
+            switch (i)
+            {
+                case 1:
+                    goto default;
+                case 2:
+                    break;
+                default:
+                    goto case 1; // FN
+            }
+        }
+
+        int Test4() => Test4(); // Noncompliant
+
+        int Test5() => 42;
+
+        int Prop
+        {
+            get // Noncompliant {{Add a way to break out of this property accessor's recursion.}}
+            {
+                return Prop;
+            }
+        }
+
+        string Prop0
+        {
+            get
+            {
+                return nameof(Prop0);
+            }
+        }
+
+        object Prop1
+        {
+            get // FN - analyzer currently only checks for property access on the 'this' object
+            {
+                return (new InfiniteRecursion())?.Prop1;
+            }
+        }
+
+        int Prop2
+        {
+            get // Not recognized, but the accessors are circularly infinitely recursive
+            {
+                (Prop2) = 10;
+                return 10;
+            }
+            set
+            {
+                var x = Prop2;
+            }
+        }
+
+        int Prop3 => 42;
+
+        int Prop5 => Prop5; // Noncompliant
+
+        int Prop6
+        {
+            get => 42;
+        }
+
+        int Prop7
+        {
+            get => Prop6; // Compliant - FP
```

Comment:
Wait, what? 😆 

Same in Sonar flavor.
```suggestion
            get => Prop6;
```